app/api/views.py

```diff
@@ -479,3 +459,32 @@ def post(self, request, *args, **kwargs):
         except IntegrityError:
             content = {'error': 'IntegrityError: you cannot create a label with same name or shortkey.'}
             return Response(content, status=status.HTTP_400_BAD_REQUEST)
+
+class CommentList(generics.ListCreateAPIView):
+    pagination_class = None
+    permission_classes = [IsAuthenticated & IsInProjectOrAdmin]
+    serializer_class = CommentSerializer
+    model = Comment
+
+    def get_queryset(self):
+        queryset = self.model.objects.filter(
+            document__project_id=self.kwargs['project_id'],
+            document=self.kwargs['doc_id']
+        ) if 'doc_id' in self.kwargs else self.model.objects.filter(
+                document__project_id=self.kwargs['project_id']
+            )
+        return queryset
+
+    def create(self, request, *args, **kwargs):
+        request.data['document'] = self.kwargs['doc_id']
```

Comment:
`request.data` should normally be immutable.
https://www.django-rest-framework.org/api-guide/serializers/#passing-additional-attributes-to-save
https://www.valentinog.com/blog/drf-request/